### PR TITLE
Feature - add debug_traceTransaction support

### PR DESCRIPTION
**Pull request by kumavis**
Adds [middleware](https://github.com/MetaMask/eth-json-rpc-middleware/blob/master/trace-transaction.js) for handling `debug_traceTransaction` calls.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "eth-hd-keyring": "^1.2.1",
     "eth-json-rpc-filters": "^1.2.5",
     "eth-json-rpc-infura": "^3.0.0",
+    "eth-json-rpc-middleware": "^1.5.0",
```
Comment (kumavis) [inline package.json]: moved out of devDeps into deps, and updated version

**Comment by kumavis** [comment]
This adds the evm, so it may increase the size of the build

**Comment by danfinlay** [comment]
@kumavis I tried using this feature, just to test it in the most basic way, but wasn't able to.

Could you give me a sample web3 command for using this? I tried, but just got an opaque error:
https://github.com/MetaMask/metamask-extension/issues/3487

**Comment by kumavis** [comment]
dan said this didn't work, so we'll need to try again

**Comment by antogh** [comment]
That's really a cool feature, keep it up with the good work!  👍

**Comment by danjm** [comment]
Closing for now. Feel free to reopen.

**Pull request closed without merging**